.github/workflows/npmpublish.yml

```diff
@@ -0,0 +1,41 @@
+name: "NPM Publish and GitHub tag"
+on:
+  push:
+    branches:
+      - main
```

Comment:
I just realized that this only runs when we push to the main branch. Is there any harm in pulling this this into `develop`? That would make sure that both the action change and code changes are pushed to `main` at the same time.